### PR TITLE
Add endpoint 5. PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by rcbcastillo**

### CODE REVIEW

**Review by Angaca — CHANGES_REQUESTED** [review]
Tests need expanding.
Invalid id
invalid body
valid body with invalid keys
valid body with valid inc_votes but invalid value

File: db/test.sql [hunk 1]
```diff
@@ -10,5 +10,5 @@
   }*/
 
 
-SELECT * FROM articles
-WHERE article_id = 1;
+SELECT * FROM articles;
```
Comment (douglashellowell) [inline db/test.sql]: try not to push these files to github!

only `git add` the correct files and/or add these files to your `.gitignore`!

File: output.txt [hunk 2]
```diff
@@ -1,6 +1,41 @@
...
...                                                                                                                                                                                body                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |     created_at      | votes
```
Comment (douglashellowell) [inline output.txt]: try not to push these files to github!

only git add the correct files and/or add these files to your .gitignore!

File: app/__tests__/app.test.js [hunk 3]
```diff
@@ -112,5 +113,84 @@ describe('GET: /api/articles/:article_id', () => {
...
+
+  test('200: responds with an error message if the resource is not found', () => {
+    return request(app)
+    .patch('/api/articles/13')
+    .send(dataToAdd)
+    .expect(200)
+    .then(({body:{message}}) => {
+      expect(message).toBe('Resource not found')
+    })
+  });
```
Comment (douglashellowell) [inline app/__tests__/app.test.js]: this is a good test but `200` is not an error status code!

You want a `404 not found` here

File: models/models.js [hunk 4]
```diff
@@ -24,6 +24,21 @@ exports.selectArticleByID = async (article_id) => {
...
+  UPDATE articles
+  SET votes = votes + $1 
+  WHERE article_id = $2
+  RETURNING *;`);
 
+  const {rows} = await db.query(queryStr, [inc_votes, article_id]);
+  const article = rows[0];
+
+  if (rows.rowCOunt === 0) {
+    return Promise.reject({status: 404, message: 'Article does not exist' })
```
Comment (douglashellowell) [inline models/models.js]: you have a typo here!

it should be `rowCount`

File: models/models.js [hunk 5]
```diff
@@ -24,6 +24,21 @@ exports.selectArticleByID = async (article_id) => {
...
   return article;
+};
 
-}
+exports.updateArticleByID = async (inc_votes, article_id) => {
+  const queryStr = format(`
+  UPDATE articles
+  SET votes = votes + $1 
+  WHERE article_id = $2
+  RETURNING *;`);
```
Comment (douglashellowell) [inline models/models.js]: `format` here isn't doing anything as you're not passing in any data for `pg-format` to use in the query!

as you are only replacing values (`$1` and `$2` replacing numbers) you don't need `pg-format` at all here!

You can just put this string in `db.query`

**Review by douglashellowell — CHANGES_REQUESTED** [review]
Nice work!

There's a mistake in your tests that needs fixing and some tweaks needed in your model

File: app/__tests__/app.test.js [hunk 6]
```diff
@@ -4,6 +4,7 @@ const request = require('supertest');
 const seed = require('../../db/seeds/seed.js');
 
 const {topicData, articleData, userData, commentData} = require('../../db/data/test-data/index.js');
+const { string } = require('pg-format');
```
Comment (emilyb93) [inline app/__tests__/app.test.js]: rogue import

File: models/models.js (hunk 4 above)
Comment (emilyb93) [reply models/models.js]: Need to fix this typo still

File: app/__tests__/app.test.js [hunk 7]
```diff
@@ -112,5 +112,84 @@ describe('GET: /api/articles/:article_id', () => {
...
+      expect(article.created_at).toBe('2020-07-09T20:11:00.000Z'),
+      expect(article.votes).toBe(101)
+    })
+  });
+
+  test('200: responds with an error message if the resource is not found', () => {
+    return request(app)
+    .patch('/api/articles/13')
+    .send(dataToAdd)
+    .expect(200)
```
Comment (Angaca) [inline app/__tests__/app.test.js]: As Doug said this should a 404

**Review by Angaca — APPROVED** [review]
Just need to fix that 200 OK to a 404 Not Found and you are good to go

**Pull request merged**